controllers/video.go

```diff
@@ -240,50 +260,92 @@ func GetManifestFile(w http.ResponseWriter, r *http.Request, db *sql.DB) {
 	client := &http.Client{}
 
 	response, err := client.Do(request)
-
 	if err != nil {
 		log.Println(err)
-		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
+		http.Error(w, "Error Streaming Video", http.StatusInternalServerError)
 		return
 	}
-
 	defer response.Body.Close()
 
 	if response.StatusCode == 404 {
-		http.Error(w, "Video record not found", http.StatusNotFound)
+		http.Error(w, "Video chunk not found", http.StatusNotFound)
 		return
 	}
 
 	bodyBytes, err := io.ReadAll(response.Body)
 
 	if err != nil {
 		log.Println(err)
-		http.Error(w, "Error retrieving video", http.StatusInternalServerError)
+		http.Error(w, "Error Streaming Video", http.StatusInternalServerError)
 		return
 	}
 
-	w.Header().Set("Content-Type", "application/x-mpegURL")
+	w.Header().Set("Content-Type", "video/MP2T")
 	w.WriteHeader(http.StatusOK)
 	w.Write(bodyBytes)
 }
 
-func GetTSFiles(w http.ResponseWriter, r *http.Request, db *sql.DB) {
-	videoName := strings.Split(r.URL.Path[1:], "/")[3]
+func DeleteVideo(w http.ResponseWriter, r *http.Request, db *sql.DB) {
+	body, err := io.ReadAll(r.Body)
 
-	videoComps := strings.Split(videoName, ".")
-	hashChecksum := sha1.New()
-	hashChecksum.Write([]byte(videoComps[0]))
-	fileId := fmt.Sprintf("%x", hashChecksum.Sum(nil))[:36]
+	if err != nil {
+		log.Println(err)
+		http.Error(w, "Error Deleting Video", http.StatusInternalServerError)
+		return
+	}
 
-	log.Println("Video chunk requested: " + fileId)
+	videoId := string(body)
```

Comment:
why adding the video ID in body?
why not in the URL path itself?
just like `GetVideo` handler?

Reply:
Just went through what came to my mind first 😄.
Changed it.